packages/subgraph/src/utils/tokens/erc1155.ts

```diff
@@ -0,0 +1,353 @@
+import {Address, BigInt, Bytes, ethereum} from '@graphprotocol/graph-ts';
+import {
+  ERC1155Balance,
+  ERC1155Contract,
+  ERC1155TokenIdBalance,
+  ERC1155Transfer
+} from '../../../generated/schema';
+import {ERC1155} from '../../../generated/templates/DaoTemplateV1_0_0/ERC1155';
+import {supportsInterface} from '../erc165';
+import {
+  DECODE_OFFSET,
+  ERC1155_safeBatchTransferFrom,
+  ERC1155_safeTransferFrom,
+  TransferType,
+  getTransferId
+} from './common';
+
+export function supportsERC1155(token: Address): bool {
+  // Double check that it's ERC1155 by calling supportsInterface checks.
+  let erc1155 = ERC1155.bind(token);
+  let introspection_01ffc9a7 = supportsInterface(erc1155, '01ffc9a7'); // ERC165
+  let introspection_d9b67a26 = supportsInterface(erc1155, 'd9b67a26'); // ERC1155
+  let introspection_00000000 = supportsInterface(erc1155, '00000000', false);
+  return (
+    introspection_01ffc9a7 && introspection_d9b67a26 && introspection_00000000
+  );
+}
+
+export function fetchERC1155(address: Address): ERC1155Contract | null {
+  // Try load entry
+  let contract = ERC1155Contract.load(address.toHexString());
+  if (contract != null) {
+    return contract;
+  }
+
+  // Detect using ERC165
+  if (!supportsERC1155(address)) {
+    return null;
+  }
+  contract = new ERC1155Contract(address.toHexString());
+  contract.save();
+
+  return contract;
+}
+
+export function updateERC1155Balance(
+  daoId: string,
+  token: string,
+  tokenId: BigInt,
+  amount: BigInt,
+  timestamp: BigInt,
+  type: TransferType
+): void {
+  // generate unique balance id
+  let balanceId = daoId.concat('_').concat(token);
```

Comment:
should we strat moving the ids to a common place?
like: https://github.com/aragon/osx-plugin-template-hardhat/blob/main/packages/subgraph/commons/ids.ts#L62
WDYT?